### PR TITLE
Add amg2023 package, remove amg package

### PR DESCRIPTION
**Pull request by becker33**
Add amg2023 package

Consolidate existing `amg` and `amg2013` packages (they reference the same code) under the `amg2013` name to minimize confusion between `amg2023` and `amg2013`.

### CODE REVIEW

**Comment by becker33** [comment]
@spackbot fix style

**Comment by becker33** [comment]
@ulrikeyang would you want to be the maintainer for this in the long term (after we get it working and into CI) or would you want someone else to manage changes to the Spack package for this?

File: var/spack/repos/builtin/packages/amg2023/package.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amg2023/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/packages/amg2013/package.py [hunk 2]
```diff
@@ -7,29 +7,23 @@
...
     homepage = "https://computing.llnl.gov/projects/co-design/amg2013"
-    url = "https://computing.llnl.gov/projects/co-design/download/amg2013.tgz"
+    git = "https://github.com/LLNL/AMG.git"
 
-    version(
-        "master",
-        sha256="b03771d84a04e3dbbbe32ba5648cd7b789e5853b938dd501e17d23d43f13c50f",
-        url="https://computing.llnl.gov/projects/co-design/download/amg2013.tgz",
-    )
+    version("develop", branch="master")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amg2013/package.py:20]: I thought we prefer the version name to match the branch name given the comparison fix some time ago.

File: var/spack/repos/builtin/packages/amg2023/package.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+class Amg2023(MakefilePackage):
+    """AMG2023 is a parallel algebraic multigrid solver for linear systems
+    arising from problems on unstructured grids. The driver provided here
+    builds linear systems for various 3-dimensional problems. It requires
+    an installation of hypre-2.27.0 or higher.
+    """
+
+    tags = ["benchmark"]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amg2023/package.py:16]: So no longer a proxy-app even though the description is the same?

File: var/spack/repos/builtin/packages/amg2023/package.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    homepage = "https://github.com/LLNL/AMG2023"
+    git = "https://github.com/LLNL/AMG2023.git"
+
+    version("develop", branch="main")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amg2023/package.py:20]: Same question here about the version-branch mismatch.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage, branch existence, and version tags though have questions about the branch naming scheme.

**Comment by tldahlgren** [comment]
Would like to hear from @rspavel about the ecp-proxy-apps now depending on `amg2013`.

**Review by junghans — DISMISSED** [review]

**Comment by junghans** [comment]
@tldahlgren @rspavel left @LANL, so we should drop him as maintainer of the proxy packages.

**Review by scheibelp — APPROVED** [review]
Reapproving after minor copyright change

**Pull request merged**